cpp/arcticdb/storage/common.hpp

```diff
@@ -91,5 +95,13 @@ struct is_key_type<entity::VariantKey> : std::true_type {};
 template <typename T>
 inline constexpr bool is_key_type_v = is_key_type<T>::value;
 
+using NativeVariantStorage = s3::S3Settings;
+using NativeVariantStorageMap = std::map<std::string, NativeVariantStorage>; //key: storage_id
+using EnvironmentNativeVariantStorageMap = std::map<std::string, NativeVariantStorageMap>; // key: env
 }  //namespace arcticdb::storage
 
+// to avoid pybind11 built-in cast as it will copy the containers
+// Below needs to be not within any namespace and before any usage of those
+PYBIND11_MAKE_OPAQUE(arcticdb::storage::NativeVariantStorageMap)
```

Comment:
We shouldn't use pybind in this layer of the code - this will make it harder to add new language bindings that don't depend on Python. I don't understand why a copy here is so bad.